src/collective/noop_communicator.h

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2022 XGBoost contributors
+ */
+#pragma once
+#include <string>
+
+#include "communicator.h"
+
+namespace xgboost {
+namespace collective {
+
+/**
+ * A no-op communicator, used for non-distributed training.
+ */
+class NoOpCommunicator : public Communicator {
```

Comment:
I see that you have already added checks for non-distributed env in various communicator implementations. Is this still necessary?

Reply:
As mentioned above, this is needed to replicate the existing rabit behavior.